pkg/controllers/pd/tasks/pod.go

```diff
@@ -265,3 +275,40 @@ func buildPDReadinessProbe(port int32) *corev1.Probe {
 		InitialDelaySeconds: defaultReadinessProbeInitialDelaySeconds,
 	}
 }
+
+func buildPDReadinessProbeWithReadyAPI(cluster *v1alpha1.Cluster, port int32) *corev1.Probe {
+	tlsClusterEnabled := coreutil.IsTLSClusterEnabled(cluster)
+
+	scheme := "http"
+	if tlsClusterEnabled {
+		scheme = "https"
+	}
+
+	readinessURL := fmt.Sprintf("%s://127.0.0.1:%d/pd/api/v2/ready", scheme, port)
+	var command []string
+	command = append(command, "curl", readinessURL,
+		// Fail silently (no output at all) on server errors
+		// without this if the server return 500, the exist code will be 0
+		// and probe is success.
+		"--fail",
+		// Silent mode, only print error
+		"-sS",
+		// follow 301 or 302 redirect
+		"--location")
+
+	if tlsClusterEnabled {
+		cacert := path.Join(v1alpha1.DirPathClusterTLSPD, corev1.ServiceAccountRootCAKey)
+		cert := path.Join(v1alpha1.DirPathClusterTLSPD, corev1.TLSCertKey)
+		key := path.Join(v1alpha1.DirPathClusterTLSPD, corev1.TLSPrivateKeyKey)
+		command = append(command, "--cacert", cacert, "--cert", cert, "--key", key)
+	}
+
+	return &corev1.Probe{
+		InitialDelaySeconds: defaultReadinessProbeInitialDelaySeconds,
+		ProbeHandler: corev1.ProbeHandler{
+			Exec: &corev1.ExecAction{
```

Comment:
Why not http probe directly ？ It means we have to ensure that `curl` exists.

Reply:
If TLS is not enabled, we can use http probe. However, when TLS is enabled, client cert must be provided and we have to use exec action.